toolkit/tools/pkg/imagecustomizerlib/cosicommon.go

```diff
@@ -16,10 +15,24 @@ import (
 	"github.com/microsoft/azurelinux/toolkit/tools/internal/safeloopback"
 )
 
+type cosiPartitionMetadata struct {
+	PartitionNum      int       `json:"partitionnum"`     // Example: 1
+	PartitionFilename string    `json:"filename"`         // Example: image_1.raw.zst
+	PartLabel         string    `json:"partlabel"`        // Example: boot
+	FileSystemType    string    `json:"fstype"`           // Example: vfat
+	PartitionTypeUuid string    `json:"parttype"`         // Example: c12a7328-f81f-11d2-ba4b-00a0c93ec93b
+	Uuid              string    `json:"uuid"`             // Example: 4BD9-3A78
+	PartUuid          string    `json:"partuuid"`         // Example: 7b1367a6-5845-43f2-99b1-a742d873f590
+	Mountpoint        string    `json:"mountpoint"`       // Example: /mnt/os/boot
+	UncompressedSize  uint64    `json:"uncompressedsize"` // Example: 104857600
+	Verity            []*Verity `json:"verity,omitempty"`
+}
+
 type ImageBuildData struct {
-	Source    string
-	KnownInfo outputPartitionMetadata
-	Metadata  *Image
+	Source       string
+	KnownInfo    cosiPartitionMetadata
+	Metadata     *Image
+	VeritySource *string
```

Comment:
If the empty string (`""`) isn't a valid value, then you should use the empty string to represent that the value is missing, instead of using `*string` and `nil`.

This helps reduce the complexity of the code.